### ISSUE TITLE
redis-cli reconnect should connect to the same DB as previous session

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#541]
```
Running redis 2.2.2, when I'm using redis-cli and the connection times out, it 
seamlessly reestablishes the connection.  However, it doesn't seamlessly select 
the DB from before the connection timeout.  On quite a few occasions I've 
incorrectly though the key was deleted, but then I recall I need to reselect 
the DB.  It's kinda tricky because I don't always know that the connection has 
been reestablished.

```

Original issue reported on code.google.com by `nirvd...@gmail.com` on 2 May 2011 at 3:28
- Merged into: #468

**Comment by GoogleCodeExporter** [comment]
```
This has been fixed in 2.2.3. redis-cli will also show the selected DB number 
in the prompt when it is non-zero.
```

Original comment by `pcnoordh...@gmail.com` on 2 May 2011 at 9:15
- Changed state: **Duplicate**

**Comment by GoogleCodeExporter** [comment]
```
Ahh.  I searched the tickets but missed this and didn't see it in the release 
notes.  I guess it's time for an upgrade.  Thanks.
```

Original comment by `nirvd...@gmail.com` on 2 May 2011 at 11:23
